### ISSUE TITLE
New Parser: exactly

### ISSUE THREAD

**Issue by jeffrose** [francisrstokes/arcsecond#46]
I found myself in a situation where based on the results of a `lookAhead()` I need to take either `x` or `x-1` characters. I couldn't find a convenient way to do that with the current API, so I wrote a new parser, `exactly`.

```javascript
exactly( 3, digit ).run( '12345' );
//=> { result: [ '1', '2', '3' ] ... }
```

Now this does open the door to some anti-patterns. While `exactly( 3, letter )` makes sense, `exactly( 3, letters )` does not.

First of all, is there a way I could've done this with the current API that I simply didn't see?
Secondly, would you want a PR for the new parser? Admittedly it's not very complicated but it works.

**Comment by francisrstokes** [comment]
Hey Jeff,

`exactly()` would be an excellent addition to the library. I've also found myself needing it, and normally i use:

```javascript
const exactly = (n, parser) => sequenceOf(Array.from({ length: n}, () => parser);
```

But it would be good to have a proper parser with descriptive error messages. So I'd welcome a PR.

As for anti-patterns - since these are already possible with things like `many(letters)` I would say it's on the library user to understand what they're constructing.

**Comment by francisrstokes** [comment]
Merged and published in 3.1.0 🎉
